django/contrib/postgres/fields/serial.py

```diff
@@ -0,0 +1,61 @@
+from django.db import connection
+from django.db.models import Field
+from django.utils.translation import ugettext_lazy as _
+
+
+class SerialField(Field):
+    description = _("Serial")
+
+    empty_strings_allowed = False
+    default_error_messages = {
+        'invalid': _("'%(value)s' value must be an integer."),
+    }
+
+    def __init__(self, *args, **kwargs):
+        kwargs['blank'] = True
+        kwargs.setdefault('unique', True)
+        super(SerialField, self).__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        kwargs.pop('blank')
+        return name, path, args, kwargs
+
+    def validate(self, value, model_instance):
+        pass
+
+    def get_db_prep_value(self, value, connection, prepared=False):
+        if not prepared:
+            value = self.get_prep_value(value)
+        return value
+
+    def get_auto_increment(self):
+        cursor = connection.cursor()
+        query = "SELECT nextval('%(db_table)s_%(field_name)s_seq')" \
+                % {
+                    'db_table': self.model._meta.db_table,
+                    'field_name': self.attname,
+                }
+        cursor.execute(query)
+        row = cursor.fetchone()
+        return row[0]
+
+    def get_prep_value(self, value):
+        value = super(SerialField, self).get_prep_value(value)
+        if value is None:
+            return None
+        return int(value)
+
+    def pre_save(self, model_instance, add):
+        value = super(SerialField, self).pre_save(model_instance, add)
+        if value is None:
+            value = self.get_auto_increment()
```

Comment:
Isn't that the receipe for a race condition where 2 values saved at the same time get the same serial value ?